code/modules/shuttle/super_cruise/shuttle_components/shuttle_console.dm

```diff
@@ -0,0 +1,358 @@
+GLOBAL_VAR_INIT(shuttle_docking_jammed, FALSE)
+
+/obj/machinery/computer/shuttle_flight
+	name = "shuttle console"
+	desc = "A shuttle control computer."
+	icon_screen = "shuttle"
+	icon_keyboard = "tech_key"
+	light_color = LIGHT_COLOR_CYAN
+	req_access = list( )
+	var/shuttleId
+
+	//For recall consoles
+	//If not set to an empty string, will display only the option to call the shuttle to that dock.
+	//Once pressed the shuttle will engage autopilot and return to the dock.
+	var/recall_docking_port_id = ""
+
+	var/request_shuttle_message = "Request Shuttle"
+
+	//Admin controlled shuttles
+	var/admin_controlled = FALSE
+
+	//Used for mapping mainly
+	var/possible_destinations = ""
+	var/list/valid_docks = list("")
+
+	//Our orbital body.
+	var/datum/orbital_object/shuttle/shuttleObject
+
+/obj/machinery/computer/shuttle_flight/Initialize(mapload, obj/item/circuitboard/C)
+	. = ..()
+	valid_docks = params2list(possible_destinations)
+	shuttleId = shuttleId
+	shuttlePortId = "[shuttleId]_custom"
+
+/obj/machinery/computer/shuttle_flight/Destroy()
+	. = ..()
+	shuttleObject = null
```

Comment:
Should this be `QDEL_NULL`?

Reply:
No, other consoles reference this object